boilerplate/openshift/osd-container-image/standard.mk

```
@@ -0,0 +1,112 @@
+# Validate variables in project.mk exist
+ifndef IMAGE_REGISTRY
+$(error IMAGE_REGISTRY is not set; check project.mk file)
+endif
+ifndef IMAGE_REPOSITORY
+$(error IMAGE_REPOSITORY is not set; check project.mk file)
+endif
+ifndef IMAGE_NAME
+$(error IMAGE_NAME is not set; check project.mk file)
+endif
+ifndef VERSION_MAJOR
+$(error VERSION_MAJOR is not set; check project.mk file)
+endif
+ifndef VERSION_MINOR
+$(error VERSION_MINOR is not set; check project.mk file)
+endif
+
+### Accommodate docker or podman
+#
+# The docker/podman creds cache needs to be in a location unique to this
+# invocation; otherwise it could collide across jenkins jobs. We'll use
+# a .docker folder relative to pwd (the repo root).
+CONTAINER_ENGINE_CONFIG_DIR = .docker
+# But docker and podman use different options to configure it :eyeroll:
+# ==> Podman uses --authfile=PATH *after* the `login` subcommand; but
+# also accepts REGISTRY_AUTH_FILE from the env. See
+# https://www.mankier.com/1/podman-login#Options---authfile=path
+export REGISTRY_AUTH_FILE = ${CONTAINER_ENGINE_CONFIG_DIR}/config.json
+# ==> Docker uses --config=PATH *before* (any) subcommand; so we'll glue
+# that to the CONTAINER_ENGINE variable itself. (NOTE: I tried half a
+# dozen other ways to do this. This was the least ugly one that actually
+# works.)
+ifndef CONTAINER_ENGINE
+CONTAINER_ENGINE=$(shell command -v podman 2>/dev/null || echo docker --config=$(CONTAINER_ENGINE_CONFIG_DIR))
+endif
+
+# Generate version and tag information from inputs
+COMMIT_NUMBER=$(shell git rev-list `git rev-list --parents HEAD | egrep "^[a-f0-9]{40}$$"`..HEAD --count)
+CURRENT_COMMIT=$(shell git rev-parse --short=7 HEAD)
+IMAGE_VERSION := $(VERSION_MAJOR).$(VERSION_MINOR).$(COMMIT_NUMBER)-$(CURRENT_COMMIT)
+
+IMAGE=$(IMAGE_REGISTRY)/$(IMAGE_REPOSITORY)/$(IMAGE_NAME)
+IMAGE_TAG=v$(IMAGE_VERSION)
+IMG?=$(IMAGE):$(IMAGE_TAG)
+IMAGE_URI=${IMG}
```

I think `IMG` has been carried through from forever ago so the consumer could do things like push to a personal quay by saying `make IMG=quay.io/me/thing:tag osd-container-image-push`. But I don't see a need to have two variables. We could do with just
```suggestion
IMAGE_URI?=$(IMAGE):$(IMAGE_TAG)
```